core/src/com/metalrain/stocksimulator/state/components/StockImpactEventComponent.java

```diff
@@ -34,14 +42,29 @@ public boolean appliesTo(String name) {
 
     public void apply(NameComponent nc, MarketDemandComponent demandComponent) {
         int index = 0;
+        String supplyHolder="";
+        String demandHolder="";
+
         for (String market_name:market_names) {
             if (market_name.equalsIgnoreCase(nc.name)) {
                 demandComponent.demand += demand_impact[index];
                 demandComponent.supply += supply_impact[index];
-                return;
+                demandHolder=(" "+market_name+":"+String.valueOf(demand_impact[index]));
+                demandViewData+=demandHolder;
+                supplyHolder=(" "+market_name+":"+String.valueOf(supply_impact[index]));
+                supplyViewData+=demandHolder;
+
             }
             index++;
+
         }
+        toString=supplyViewData+demandViewData;
 
     }
+    @Override
+    public String toString(){
+        return toString;
```

Comment:
Generate a string in this function, iterate over market names, and then generate a string like this.

ABC:+3, XYZ:-2, QWE:1

Basically, MARKET_NAME: (Supply - Demand), ....